### ISSUE TITLE
Request for exceptions usage documentation

### ISSUE THREAD

**Issue by ahmednofal** [jbeder/yaml-cpp#1122]
Thank you for the work on this great project. I wanted to ask you for documentation ( I could not find on my own ) for exceptions' specific usage. 

For instance it is not clear which exceptions are propagated up to the user's logic and which are consumed internally. I found this [file](https://github.com/jbeder/yaml-cpp/blob/master/src/exceptions.cpp) and it seems theses are the exceptions the user can expect to catch from yaml-cpp. However, I am not completely sure. 

There are certain types of exceptions I could not manually force yaml-cpp to throw. 
For instance the YAML::KeyNotFound exception is not thrown when you do this: 
```
YAML::Node node_representing_yaml_config = YAML::LoadFile("example.yaml");
node_representing_yaml_config["notpresentkey"];
```
TL;DR

What are the exceptions the user can expect yaml-cpp to throw and when each of them is thrown (logically/semantically) ?
